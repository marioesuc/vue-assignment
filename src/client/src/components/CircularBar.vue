<template>
  <div id="circularbar">
    <span id="currentValue">{{`${currentValue} ${unit}`}}</span>
    <VueSvgGauge
      :start-angle="-110"
      :end-angle="110"
      :value="currentValue"
      :separator-step="0"
      :min="0"
      :max="maxValue"
      :gauge-color="[{ offset: 0, color: '#347AB0'}, { offset: 100, color: '#8CDFAD'}]"
      :scale-interval="0.1"
      :transition-duration="0"
    />
  </div>
</template>

<script>
import { VueSvgGauge } from 'vue-svg-gauge'

export default {
  name: 'CircularBar',
  components: {
    VueSvgGauge,
  },
  props: {
    currentValue: Number,
    maxValue: Number,
    unit: String,
  },
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
#circularbar {
  margin: 20px;
  width: 300px;
  height: 200px;
  text-align: center;
}
#currentValue {
  position: relative;
  top: 160px;
  color: green;
  font-size: 30px;
  font-weight: bold;
  text-align: center;
  margin: 0 auto;
}
</style>
