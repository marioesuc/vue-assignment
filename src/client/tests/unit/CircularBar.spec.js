import { shallowMount } from "@vue/test-utils";
import CircularBar from "@/components/CircularBar.vue";
import { VueSvgGauge } from "vue-svg-gauge";

describe("CircularBar.vue", () => {
  it("renders correctly", () => {
    const propsData = {
      title: "title",
      currentValue: 15,
      maxValue: 100,
      unit: "km/h"
    };

    const wrapper = shallowMount(CircularBar, { propsData });
    expect(wrapper.find(VueSvgGauge).exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
