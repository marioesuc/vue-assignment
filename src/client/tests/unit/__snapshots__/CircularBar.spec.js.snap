// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularBar.vue renders correctly 1`] = `
<div id="circularbar"><span id="currentValue" class="thresholdNotExceeded">15 km/h</span> <span id="title">title</span>
  <vuesvggauge-stub value="15" min="0" max="100" startangle="-110" endangle="110" innerradius="60" separatorstep="0" separatorthickness="4" gaugecolor="[object Object],[object Object]" basecolor="#DDDDDD" easing="Circular.Out" scaleinterval="0.1" transitionduration="0"></vuesvggauge-stub>
</div>
`;
